examples/solvers/find_optimal_parameters.py

```diff
@@ -0,0 +1,234 @@
+"""
+Example of how to use optimization in order to pick reconstruction parameters.
+
+In this example, we solve the tomographic inversion problem with different
+regularizers (fbp, huber and tv) and pick the "best" regularization parameter
+for each method w.r.t. a set of reference data.
+
+To find the "best" parameter we use Powells method to optimize a figure of
+merit, here the L2-distance to the true result.
+"""
+
+import numpy as np
+import odl
+import scipy
+
+
+def optimal_parameters(reconstruction, fom, phantoms, data,
+                       initial_param=0):
+    """Find the optimal parameters for a reconstruction method.
+
+    Notes
+    -----
+    For a forward operator :math:`A : X \to Y`, a reconstruction operator
+     parametrized by :math:`\theta` is some operator :math:`R_\theta : Y \to X`
+     such that
+
+    .. math::
+        R_\theta(A(x)) \approx x.
+
+    The optimal choice of :math:`\theta` is given by
+
+    .. math::
+        \theta = \argmin_\theta fom(R(A(x) + noise), x)
+
+    where :math:`fom : X \times X \to \mathbb{R}` is a Figure of Merit.
+
+
+    Parameters
+    ----------
+    reconstruction : callable
+        Function that takes two parameters:
+
+            * data : The data to be reconstructed
+            * parameters : Parameters of the reconstruction method
+
+        The function should return the reconstructed image.
+    fom : callable
+        Function that takes two parameters:
+
+            * reconstructed_image
+            * true_image
+
+        and returns a scalar Figure of Merit.
+    phantoms : sequence
+        True images.
+    data : sequence
+        The data to be reconstructed.
```

Comment:
I know this is a very widely used term, to "reconstruct the data", but IMO it's wrong because it's not the data that is reconstructed, but the unknown quantity ("volume" or "image"). Should rather be "data to reconstruct from" or so.